src/unix/solarish/compat.rs

```diff
@@ -169,3 +170,33 @@ pub unsafe fn forkpty(
 
     0
 }
+
+pub unsafe fn getpwent_r(
+    pwd: *mut passwd,
+    buf: *mut ::c_char,
+    buflen: ::size_t,
+    result: *mut *mut passwd,
+) -> ::c_int {
+    let res = native_getpwent_r(pwd, buf, buflen.try_into().unwrap());
+    if res.is_null() {
+        -1
+    } else {
+        *result = res;
+        0
+    }
+}
+
+pub unsafe fn getgrent_r(
+    grp: *mut ::group,
+    buf: *mut ::c_char,
+    buflen: ::size_t,
+    result: *mut *mut ::group,
+) -> ::c_int {
+    let res = native_getgrent_r(grp, buf, buflen.try_into().unwrap());
+    if res.is_null() {
+        -1
+    } else {
+        *result = res;
+        0
+    }
```

Comment:
Same sort of change required for this, as the convention with `getgrent_r()` is the same:

```
       Applications wishing to check for error situations should set errno to
       0 before calling getgrnam(), getgrnam_r(), getgrent(),
       getgrent_r()getgrgid(), getgrgid_r(), fgetgrent(), and fgetgrent_r().
       If these functions return a null pointer and errno is non-zero, an
       error occurred.
```